File: dotnet/Lexy.Poc/Lexy.Poc.Core/Language/ScenarioName.cs
using System;
using System.Collections.Generic;
using Lexy.Poc.Core.Parser;

namespace Lexy.Poc.Core.Language
{
    public class ScenarioName : Node
    {
        public string Value { get; private set; } = Guid.NewGuid().ToString("D");

        public void ParseName(string parameter)
        {
            Value = parameter;
        }

        public ScenarioName(SourceReference reference) : base(reference)
        {
        }

        public override string ToString() => Value;

        protected override IEnumerable<INode> GetChildren()
        {
            yield break;
        }

        protected override void Validate(IValidationContext context)
        {
        }
    }
}

File: typescipt/src/language/scenarios/scenarioName.ts


export class ScenarioName extends Node {
   public string Value { get; private set; } = Guid.NewGuid().toString(`D`);

   public ScenarioName(SourceReference reference) {
     super(reference);
   }

   public parseName(parameter: string): void {
     Value = parameter;
   }

   public override toString(): string {
     return Value;
   }

   public override getChildren(): Array<INode> {
     yield break;
   }

   protected override validate(context: IValidationContext): void {
   }
}
